### PR TITLE
Product of maximum in first array and minimum in second

### PR DESCRIPTION
**Pull request by jtomar24**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

added code for #4579 issue .please review this and let me know if changes required .

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by jtomar24** [comment]
I will add logic in the function.

**Comment by jtomar24** [comment]
I had added the logic  in the function.

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*Given two arrays, the task is to calculate the product of max element of first array and min element of second array*/
+
+/*Given two arrays, the task is to calculate the product of max element of first array and min element of second array*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Remove the duplicate description and write bit more description

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+#include<bits/stdc++.h>
+using namespace std;
+
+int find_product(int a[], int b[], int n, int m)
+    {
+        //main logic
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Remove redundant comments

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+int find_product(int a[], int b[], int n, int m)
+    {
+        //main logic 
+        int num1=*max_element(a, a+ n);
+        int num2=*min_element(b, b+ m);
+        int r=num1*num2;
+        return r;
+    }
+//main function
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Remove redundant comments

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+int main()
+{
+    //initialising and declaring two array 
+    int a[] = { 7, 3, 5, 8, 4};
+    int b[] = { 9, 8, 6, 4, 2};
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Don't use static inputs use dynamic inputs with proper display message.

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    
+    //calculating the number of elements in the two array
+    int n = sizeof(a) / sizeof(a[0]);
+    int m = sizeof(b) / sizeof(b[0]);
+    int prod=find_product(a,b,n,m);
+    cout<<prod;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Don't use single letters use proper naming conventions for variables

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Comment by jtomar24** [comment]
Sir could you tell me how to resolve this ?I m new to open source I don't know how to deal with this problem.

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
+/*Given two arrays:arr1[] and arr2[] of different size i.e.,no_of_ele_in_arr1 amd no_of_ele_in_arr2 respectively.
+Task :calculate the product of max element of first array i.e.,arr1[] and min element of second array i.e.,arr2[]
+Example:we are given two arrays arr1={1,2,3,4} and arr2={5,6,7,8,9} ,so the maximum element in array arr1 is 4 and minimum element in array arr2 is 5.Hence,the product of max element in array arr1 and min element in array arr2 is 4*5=20.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Can you make it short, crisp and easy to read? Don't give examples here. Keep it at the bottom

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+Approach:
+->take two arrays as input.
+->using std library of cpp we will find the maximum element in array arr1 and minimum element in array arr2
+->after getting the max and min element we will multiply them.
+->return the product.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Use bulleted list like `-` to denote points, not `->`

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+enter the elements of array 2:
+9, 8, 6, -4, 2
+
+Sample Output2 :
+product of max element of first array and min element of second array: -28 
+ 
+Time Complexity : O(n)
+Auxiliary Space : O(1) 
+*/
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Add a new line at the EOF

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+enter the elements of array 1:
+7, -3, 5, -8, 4
+enter the no. of elements in array 2:
+5
+enter the elements of array 2:
+9, 8, 6, -4, 2
+
+Sample Output2 :
+product of max element of first array and min element of second array: -28 
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Remove this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Use cpplint It will guide you how to format code properly for reference use this one [here](https://cpplint.appspot.com/).

**Comment by jtomar24** [comment]
I had made most of the changes which were listed regarding whitespaces .

File: C-Plus-Plus/README.md [hunk 10]
```diff
@@ -20,6 +20,7 @@
 - [All Possible Sum Using FFT](cp/all_possible_sums_using_fft.cpp)
 - [Binary_Coded_String](cp/Binary_Coded_String.cpp)
 - [Check Anagrams](cp/check_anagrams.cpp)
+- [product of max in first array and min in second](cp/product_of_max_in_first_array_and_min_in_second.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Keep it `Product of the maximum element and the minimum element in two consecutive arrays`

File: C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp [hunk 11]
```diff
@@ -0,0 +1,57 @@
+/*Given two arrays:arr1[] and arr2[] of different size i.e.,no_of_ele_in_arr1 amd no_of_ele_in_arr2 respectively.
+Task :calculate the product of max element of first array i.e.,arr1[] and min element of second array i.e.,arr2[]
+
+Approach:
+-take two arrays as input.
+-using std library of cpp we will find the maximum element in array arr1 and minimum element in array arr2
+-after getting the max and min element we will multiply them.
+-return the product.
+*/
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/product_of_max_in_first_array_and_min_in_second.cpp]: Can you try to follow the standards and not make any shortforms?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Kindly use Proper formatting.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**